### PR TITLE
Split out get_events and co into a worker store

### PR DESCRIPTION
**Pull request by erikjohnston**
The diff for `synapse.storage.events` is a bit silly, all I've done is move the functions up the file.

### CODE REVIEW

File: synapse/storage/events.py [hunk 1]
```diff
@@ -199,1467 +199,1472 @@ def f(self, *args, **kwargs):
     return f
 
 
-class EventsStore(SQLBaseStore):
-    EVENT_ORIGIN_SERVER_TS_NAME = "event_origin_server_ts"
-    EVENT_FIELDS_SENDER_URL_UPDATE_NAME = "event_fields_sender_url"
-
+class EventsWorkerStore(SQLBaseStore):
```
Comment (richvdh) [inline synapse/storage/events.py]: shouldn't _event_persist_queue be in the EventsStore?

File: synapse/storage/events.py [hunk 2]
```diff
@@ -199,1467 +199,1472 @@ def f(self, *args, **kwargs):
...
-        logger.debug("calling resolve_state_groups from preserve_events")
-        res = yield self._state_resolution_handler.resolve_state_groups(
-            room_id, state_groups, events_map, get_events
+class EventsStore(EventsWorkerStore):
+    EVENT_ORIGIN_SERVER_TS_NAME = "event_origin_server_ts"
+    EVENT_FIELDS_SENDER_URL_UPDATE_NAME = "event_fields_sender_url"
+
+    def __init__(self, db_conn, hs):
+        super(EventsStore, self).__init__(db_conn, hs)
+        self._clock = hs.get_clock()
```
Comment (richvdh) [inline synapse/storage/events.py]: this appears to be redundant, fwiw.

**Comment by richvdh** [comment]
oh, I wonder if it would be worth putting EventsWorkerStore in a separate python file. events.py is pretty huge.

**Comment by erikjohnston** [comment]
I've split out the events into two files, do you want to take a quick look before I merge?

**Pull request merged**